snappy_wrappers/wrappers/somatic_variants_checking/summarize-vcf.py

```diff
@@ -0,0 +1,254 @@
+# import packages
+import argparse
+import gzip
+import json  # for writing out the json file
+
+from cyvcf2 import VCF  # for reading vcf file
+import tabix
+
+parser = argparse.ArgumentParser(description="Gathering information of variants")
+parser.add_argument(
+    "--rawvcf",
+    help="raw vcf file",
+)
+parser.add_argument(
+    "--filtered-vcf",
+    help="filtered vcf file",
+)
+parser.add_argument(
+    "--exom-bedfile",
+    help="exom regions bed file",
+)
+parser.add_argument("--padding", help="it's padding", default=0)
+parser.add_argument(
+    "--ignore-regions",
+    help="bed file contains repeated or difficult to map regions",
+)
+parser.add_argument(
+    "-m",
+    "--minimal",
+    type=int,
+    default=1,
+    help="threshold for defining a variant that has minimal support reads",
+)
+parser.add_argument(
+    "-l",
+    "--limited",
+    type=int,
+    default=5,
+    help="threshold for defining a variant that has limited support reads",
+)
+parser.add_argument(
+    "--output",
+    help="json file contains information about vcf file",
+)
+args = parser.parse_args()
+
+
+def check_variant_in_bed(v_chrom, v_start, v_end, bed_intervals, padding=0):
+    try:
+        tmp = []
+        records = bed_intervals.query(v_chrom, v_start - padding, v_end + padding)
+        for record in records:
+            tmp.append(record)
+        if len(tmp) >= 1:
+            return True
+        else:
+            return False
+    except OSError:
+        print("You should index the bed file with tabix first!")
+
+
+def get_contigs_from_bed_file(bedfile):
+    contigs = []
+    try:
+        with gzip.open(bedfile, "r") as intervals:
+            for interval in intervals:
+                line = interval.decode()
+                tmp = line.split("\t")[0]
+                if tmp not in contigs:
+                    contigs.append(tmp)
+        return contigs
+    except OSError:
+        print("Error reading the GZIP file.")
+
+
+def get_variant_type(ref, alt):
+    if (len(alt) == 1) and (len(ref) == len(alt)):
+        return "snp"
+    elif len(alt) != len(ref):
+        return "indels"
+
+
+def check_sp_read(variant, minimal, limited):
```

Comment:
That's wrong: you are checking the depth of the read, presumably for both normal and tumor samples. You need to check the `AD` field in the genotype (`FORMAT`) information. For each variant, the `AD` field is (probably) stored as a list, because it's number is `R` (check out the [vcf file format](https://samtools.github.io/hts-specs/VCFv4.3.pdf)). You need to make sure that you are picking the right columns. For that you may need to know the the library name.